tests/test_completable_pair_gate.py

```
@@ -0,0 +1,176 @@
+    def test_zero_still_disables_the_cap(self):
+        with mock.patch.dict(os.environ, {"HUNTER_COMPLETABLE_CAP": "0"}):
+            assert load().max_completable_pair_cost == 0.0
+
+    def test_a_dollar_is_still_accepted(self):
+        with mock.patch.dict(os.environ, {"HUNTER_COMPLETABLE_CAP": "1.0"}):
+            assert load().max_completable_pair_cost == 1.0
+
+    @pytest.mark.parametrize("bad", ["nan", "inf", "-0.01", "1.5", "abc"])
+    def test_a_non_finite_negative_or_oversized_dead_band_is_refused(self, bad):
+        with mock.patch.dict(os.environ, {"HUNTER_REQUOTE_DEAD_BAND": bad}):
+            with pytest.raises(ValueError, match="HUNTER_REQUOTE_DEAD_BAND"):
```

Ruff (0.16.2)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)